java/entrypoint.sh

```diff
@@ -32,22 +32,37 @@ if [ "${FORGE_COMPATIBILITY}" = 1 ] && [ -z "${JLINE_ARGS}" ] && [ -z "${FORGE_V
 fi
 
 # Log4j2 vulnerability workaround
-if [ "${LOG4J2_VULN_WORKAROUND}" = 1 ] && [ -z "${FORGE_VERSION}" ]; then
-    MODIFIED_STARTUP=$(echo "${MODIFIED_STARTUP}" | sed -E 's/-Xmx([0-9]+)M/& -Dlog4j2.formatMsgNoLookups=true/')
-    echo -e "\033[1;33mNOTE: \033[0mThe Log4j2 vulnerability workaround has been enabled. If you're running an unpatched server software build, remember to update ASAP as this workaround may be removed at any time, and is not effective in older versions of the game."
+if [ ! "${LOG4J2_VULN_WORKAROUND}" = "disabled" ] && [ -n "${LOG4J2_VULN_WORKAROUND}" ]; then
+    case "${LOG4J2_VULN_WORKAROUND}" in
+    "Enable for 1.17+")
+        MODIFIED_STARTUP=$(echo "${MODIFIED_STARTUP}" | sed -E 's/-Xmx([0-9]+)M/& -Dlog4j2.formatMsgNoLookups=true/')
+        echo -e "\033[1;33mNOTE: \033[0mThe Log4j2 vulnerability workaround for 1.17 and 1.18 has been enabled. Please consider updating your server software if you haven't already."
+        ;;
+
+    "Enable for 1.12 - 1.16")
+        if [ ! -f "log4j2_112-116.xml" ]; then
+            curl --silent -Lo log4j2_112-116.xml https://launcher.mojang.com/v1/objects/02937d122c86ce73319ef9975b58896fc1b491d1/log4j2_112-116.xml
+        fi
+        MODIFIED_STARTUP=$(echo "${MODIFIED_STARTUP}" | sed -E 's/-Xmx([0-9]+)M/& -Dlog4j.configurationFile=log4j2_112-116.xml/')
+        echo -e "\033[1;33mNOTE: \033[0mThe Log4j2 vulnerability workaround for 1.12 - 1.16.5 has been enabled. Please consider updating your server software if you haven't already."
+        ;;
+
+    "Enable for 1.7 - 1.11")
+        if [ ! -f "log4j2_17-111.xml" ]; then
+            curl --silent -Lo log4j2_17-111.xml https://launcher.mojang.com/v1/objects/4bb89a97a66f350bc9f73b3ca8509632682aea2e/log4j2_17-111.xml
+        fi
+        MODIFIED_STARTUP=$(echo "${MODIFIED_STARTUP}" | sed -E 's/-Xmx([0-9]+)M/& -Dlog4j.configurationFile=log4j2_17-111.xml/')
+        echo -e "\033[1;33mNOTE: \033[0mThe Log4j2 vulnerability workaround for 1.7 - 1.11.2 has been enabled. Please consider updating your server software if you haven't already."
+        ;;
+    esac
 fi
 
 # SIMD operations (https://github.com/sparkedhost/images/issues/4)
-if [ "${SIMD_OPERATIONS}" = 1 ] && [ -z "${FORGE_VERSION}" ]; then
```

Comment:
This and similar needs adding back else messages will be shown in the console when it's not actually enabled